transport/peeragent.go

```diff
@@ -0,0 +1,19 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peeragent.go -package=transporttest go.uber.org/yarpc/transport PeerAgent,PeerSubscriber
+
+// PeerAgent manages Peers across different PeerSubscribers.  A PeerSubscriber will request a Peer for a specific
+// PeerIdentifier and the PeerAgent has the ability to create a new Peer or return an existing one.
+type PeerAgent interface {
```

Comment:
Let’s just call this Agent. Let’s optimize for succinct unless there’s potential for ambiguity. In this case, I expect that there will be no other kind of agent.

In the doc, note that the Agent is scoped to a kind of transport, so for any number of HTTP Outbounds, there would be one shared HTTPAgent.